src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/Controllers/TeamPlanController/TeamPlanController.tsx

```diff
@@ -1,30 +1,100 @@
-import { FieldValues, UseFormRegister } from 'react-hook-form'
+import { UseFormRegister, UseFormSetValue } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
 
-import { MIN_NB_SEATS_PRO } from 'shared/utils/upgradeForm'
+import {
+  IndividualPlan,
+  useAccountDetails,
+  useAvailablePlans,
+} from 'services/account'
+import {
+  canApplySentryUpgrade,
+  findProPlans,
+  findSentryPlans,
+} from 'shared/utils/billing'
+import {
+  MIN_NB_SEATS_PRO,
+  TEAM_PLAN_MAX_ACTIVE_USERS,
+  UPGRADE_FORM_TOO_MANY_SEATS_MESSAGE,
+} from 'shared/utils/upgradeForm'
 import TextInput from 'ui/TextInput'
 
 import BillingOptions from './BillingOptions'
 import PriceCallout from './PriceCallout'
 import UserCount from './UserCount'
 
 import { NewPlanType } from '../../constants'
+import { UpgradeFormFields } from '../../UpgradeForm'
+
+interface Errors {
+  seats?: {
+    message?: string
+  }
+}
+
+interface ErrorBannerProps {
+  errors?: Errors
+  setFormValue: UseFormSetValue<UpgradeFormFields>
+  setSelectedPlan: (plan: IndividualPlan) => void
+}
+
+function ErrorBanner({
+  errors,
+  setFormValue,
+  setSelectedPlan,
+}: ErrorBannerProps) {
+  const { provider, owner } = useParams<{ provider: string; owner: string }>()
+  const { data: plans } = useAvailablePlans({ provider, owner })
```

Comment:
Something about calling all these hooks for the error banner almost makes me think we should be splitting this into a separate file to clean up some of the "distractions" from what the TeamPlanController is doing. 

How it currently is is fine though if that proves to be a bigger effort